### ISSUE TITLE
Many Aw, snap! pages

### ISSUE THREAD

**Issue by teran-mckinney** [gliaskos/freebsd-chromium#14]
Hi everyone,

First off, thank you for porting this over. I'm sure it's a rather tricky one to get right and generally Chrome has been very stable for me on FreeBSD. That's at least a year or so, until now.

Since the latest pkg upgrade I've been having continual Aw, snap! errors. Happens on LinkedIn, HipChat, and others. They range from 9/10 to maybe 1/50 page loads. I know little about them and am not sure if this is the place to report it, but am hoping you guys see something similar and know what's going on, or have suggestions on how I can best debug it.

My Chromium version is 43.0.2357.65. I'm on FreeBSD 10.1-RELEASE, x86_64.

Thank you,
Teran

**Comment by rene0** [comment]
With Chromium 43 I get a segmentation fault (but without a coredump or debug info or an "Aw snap!" page) when goign to gmail.com...

**Comment by LeFroid** [comment]
Same here -- I've been trying to figure out the source of these errors to no avail.
There also seems to be a crash when visiting chrome://settings , although the other chrome:// pages work just fine.

I'm currently looking through the commits between chromium 42 and 43, although there are over 1,000 changes to check one-by-one

**Comment by lapo-luchini** [comment]
Well, 1000 commits are but 10 builds with a dichotomic search. Just kidding. ;)
I'll keep on 42 for now…

**Comment by sysconfig** [comment]
Having the same issue as rene0:  Segfault when visiting chrome://settings, other chrome:// URLs working just fine.

This still happens with the latest version from last night: 43.0.2357.81 (SVN ports/head revision r388285)

Clearing ``.cache/chromium`` and ``.config/chromium`` didn't do anything to resolve this.

```
$ uname -a

FreeBSD monster 10.1-STABLE FreeBSD 10.1-STABLE #0 r283875: 
   Mon Jun  1 11:24:39 BST 2015     root@monster:/usr/obj/usr/src/sys/MONSTER  amd64

(This is indeed a GENERIC kernel minus Witness/Debugging, 
   SVN base/stable/10 revision r283875)


$ chrome --diagnostics

[PASS] Install type
       System Level

[PASS] Chrome version test
       43.0.2357.81

[PASS] User data path
       Path exists and is writable: 17533404

[PASS] Local state path
       Path exists and is writable: 1784

[PASS] App dictionaries directory path
       Path exists

[PASS] Resources path
       Path exists

[PASS] Available disk space
       Free space: 364336414720

[PASS] User preferences integrity
       File parsed OK

[PASS] Local state integrity
       File parsed OK

[PASS] Bookmark file
       File not found (but that is OK)

[PASS] Web Data database
       No corruption detected

[PASS] Cookie database
       No corruption detected

[PASS] History database
       No corruption detected

[PASS] Thumbnails database
       File not found (but that is OK)

[PASS] Database tracker database
       No corruption detected
```

**Comment by teran-mckinney** [comment]
I'm still seeing this on 43.0.2357.81 as well. It seems like the Aw, snap! pages come in waves and a ton of tabs crash at once. I can reproduce it every since time on this page: http://decentraland.org/app/

Downloads don't really seem to work. Additionally, the Chrome Settings page is an instant crash where the browser dies.

**Comment by ivadasz** [comment]
My pull-request seems to fix at least the http://decentraland.org/app/ page for me:
https://github.com/gliaskos/freebsd-chromium/pull/19

**Comment by teran-mckinney** [comment]
@ivadasz fixes chrome://settings and http://decentraland.org/app/. Thank you so much!

I'll keep using it to see if anything else comes up.

Where do we stand with #21 if this one fixes it? I am currently using the chromium available in the pkg repository:

$ pkg info chromium
chromium-43.0.2357.130
Name           : chromium
Version        : 43.0.2357.130
Installed on   : Sun Jun 28 18:54:42 UTC 2015
Origin         : www/chromium
....

**Comment by teran-mckinney** [comment]
Looks like some pages still crash. I don't get the one to many crashing effect as before, but it's generally much better. Reloading usually fixes it, except on certain pages. Some tabs will also get completely hung.

**Comment by rene0** [comment]
It looks like all pages mentioned here are loadable again?

**Comment by teran-mckinney** [comment]
Yes, it seems to be the case. Aside from the Chrome plugin download page crash mentioned in #23, it's been crash free with the last update. But, not tested for very long yet.

Downloads may be broken again, not certain on that yet.

**Comment by rene0** [comment]
True, downloads segfault if you right-click on the green progress clock :(

**Comment by teran-mckinney** [comment]
I had some more aww, snap pages. In some scenarios one page will crash and others with it.

**Comment by teran-mckinney** [comment]
I get a consistent endless loading issue, not an aww, snap, on this page: https://www.digicert.com/welcome/ssl-plus.htm

Can you confirm the same?

Thank you!

**Comment by LeFroid** [comment]
Just as a heads up, I'm currently testing a few patches that should address the download issue and should be sending a PR in the next couple of days. There are a few files from OpenBSDs port that should cover this

**Comment by rene0** [comment]
Patch for download issue committed to the Ports repository (head), not yet to the quarterly Ports branch.

**Comment by teran-mckinney** [comment]
Hi,

I'm still seeing these Aw, snap! pages on FreeBSD 10.2. Do you know what's causing them?

Thank you,
Teran

**Comment by ameiji** [comment]
I can also confirm that the issue is still present for chromium 47.0.2526.80. (10.2-RELEASE-p7)

**Comment by nforced** [comment]
Same here, can't use Facebook because of too frequent crashes!
pkg info chromium
chromium-47.0.2526.106
Name           : chromium
Version        : 47.0.2526.106

FreeBSD 10.2-RELEASE-p7

**Comment by yonas** [comment]
Chromium is becoming unusable to me....

**Comment by z2v** [comment]
Same here. 

chromium-47.0.2526.106
Name           : chromium
Version        : 47.0.2526.106
Installed on   : Thu Jan 14 21:53:27 2016 MSK
Origin         : www/chromium
Architecture   : freebsd:11:x86:64

**Comment by aaron-baugher** [comment]
Same here, at least 10% failure rate with 47.0.2526.106 (64-bit) on 10.2 amd64.

I've been sticking with 42 (42.0.2311.135), which is completely solid, because every version since has had this problem.  Oddly, many times the page will crash immediately when I click on something, too quickly to be caused by something which is being downloaded and displayed or executed.  Just click-crash.  Other times it doesn't happen until the page is partially loaded.

I was okay with staying with version 42.  Unfortunately, Google changed something so that that version can no longer access google.com sites via SSL.  That makes all their sites unusable, even for logging into other sites using a Google+ account.  I don't know whether it's a legitimate bug or a way to force everyone to upgrade, but 47 fixes that.  So I'm trying to put up with the crashes with 47, but I don't know if that's going to be possible.

**Comment by aaron-baugher** [comment]
If it helps: sometimes even the 'New Tab' page crashes immediately before clicking anything, with no extensions installed.  So whatever the problem is, that page can trigger it, and it doesn't require any bad scripts downloaded for a remote page.  Though complex sites like Facebook do seem to crash more often.

**Comment by teran-mckinney** [comment]
I've seen the 'new page tabs' crash as well. I think I've also turned off Javascript and had crashes.

It may be something to do with a failure rate per object and some pages having huge amounts of objects. Craigslist is actually pretty crashy for me, which is not what I'd normally expect.

**Comment by teran-mckinney** [comment]
I was talking to a colleague last night. He said that Microsoft pushed a Windows 10 update that broke Chromium and would only run with `--no-sandbox`. I tested with the flag last night and had almost no crashes, but progressively more towards the end.

This morning I decided to try a pseudo-scientific check of loading X pages on Facebook and hoping for a crash. It didn't crash immediately like I had hoped. If it did, I could easily say that test X gave Y crashes versus Z crashes with `--no-sandbox`. Oh well. The infrequency of it really drives me crazy.

That said, maybe some of you can test with `--no-sandbox`? What's apparent is that it isn't perfect. Maybe it's a 200% improvement, which would be nice to know and maybe help narrow down the bug.

It'd be really great if we had some automated testing with Chrome and maybe Selenium. If we could automate the crashes it might be faster to tweak things and narrow down what's causing them (it's in the renderer, right?).

Thank you all for your feedback here. I'm susprised we haven't had any breakthroughs yet, but at least awareness about this has gone up.

**Comment by AusMatt** [comment]
I've just returned to using my FreeBSD NAS as a desktop after my Windows laptop failed.
I'm using Chromium 49.0.2623.112 (64-bit) on a recently updated 10-STABLE system.
> uname -a
FreeBSD nas 10.3-STABLE FreeBSD 10.3-STABLE #58 r297922: Wed Apr 13 23:53:13 ACST 2016

I'm getting a lot of "Aw, snap!" pages tonight... and it's really annoying!
examples are:
- Newegg
- Google Drive

I'd like to think that we can methodically investigate this somehow with an automated testing suite on Chromium builds that are bisected to narrow down problematic commits.
Has anyone made any progress on this?

I'm a big Google Chrome user and wouldn't like to have to go back to Firefox on my FreeBSD machine.

**Comment by puma-ac** [comment]
A "work around" for the issue of continuous "aw snap" pages in chromium is to ...
go to advanced settings
privacy > content settings > cookies
select > Keep local data only until you quit the browser
Exit the program and trigger the removal of all the website data.
You should have fresh data each time you launch Chromium.  
I will be using this setting until the issue is resolved by the developers.

**Comment by yonas** [comment]
@puma-ac Any idea why that helps?

**Comment by lf94** [comment]
I am having the same issue on Chromium 49.

**Comment by teran-mckinney** [comment]
@puma-ac, I don't think that will help, but I am going to give it a try. I setup Chromium with master_preferences and set it as a default. When I got it working, on the first run the Chrome settings page crashed.

I think the issue is in the renderer. It's possible that some of the crashes will be mitigated, but I think it will mostly be business as usual.

I've found Chrome to be stable for hours, or crashing right off the bat. I've often found one setting will "seem" to fix it, until a couple hours later.

Quite curious to try Chromium 50, once it's ready.

**Comment by danlamanna** [comment]
Same issue, `Chromium 49.0.2623.112 `.

**Comment by nforced** [comment]
freebsd-chromium is DOOMed, this is going for years by now...

**Comment by yonas** [comment]
@nforced That's pretty exaggerated. I'm using Chromium every day and pages crash infrequently. We just have to figure out why it sometimes crashes more often.

From a developers point of view, technically it shouldn't crash at all. I suspect these crashes are related to the same root issue(s), so finding the source of one crash might be enough to solve a majority of these crashes on FreeBSD.

Perhaps running `ktrace chrome` to launch chromium, and then `kdump` to look at the output of ktrace.out would help.

**Comment by teran-mckinney** [comment]
Same issue with 51.0.2704.84_1 on 10.3, x86_64.

**Comment by lf94** [comment]
So twice over the past few months I've run ```ktrace chrome``` and oddly enough, chromium won't crash. What could be the reason for this? I really want to solve this issue.

**Comment by lf94** [comment]
I get this in a terminal when the Aw Snap! appears:

Received signal 11 SEGV_MAPERR 000000000000
#0 0x0000008b4bca <unknown>
#1 0x00080b007b37 <unknown>
#2 0x00080b00722c <unknown>
[end of stack trace]

Can these be mapped to real functions? Is the freebsd chrome not compiled with debug symbols?
